estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,63 @@
+from odoo import models, fields, api
+from datetime import timedelta
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'Property Offer'
+    _order = 'price desc'
+
+    price = fields.Float(string="Price")
+    status = fields.Selection([('accepted', 'Accepted'), ('refused', 'Refused')], string="Status", copy=False)
+    partner_id = fields.Many2one('res.partner', string="Partner", required=True)
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+    property_id = fields.Many2one('estate.property', 'Property', required=True, ondelete="cascade")
+    property_type_id = fields.Many2one(related="property_id.property_type_id")
+
+    @api.depends('create_date', 'validity')
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = record.create_date + timedelta(days=record.validity)
+            else:
+                record.date_deadline = fields.Date.today() + timedelta(days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            if record.date_deadline and record.create_date:
+                create_date_date = record.create_date.date()
+                record.validity = (record.date_deadline - create_date_date).days
+            else:
+                record.validity = 0
```

Comment:
```suggestion
          record.validity = (record.deadline_date - record.property_id.create_date.date()).days
```
Can you give it a try? Make sure to check all possible cases. 
same for compute also